pallene/to_ir.lua

```diff
@@ -820,9 +876,20 @@ function ToIR:exp_to_assignment(cmds, dst, exp)
 
     elseif tag == "ast.Exp.Lambda" then
         local f_id = self:register_lambda(exp, "$lambda")
+        local func = self.module.functions[f_id]
         self:convert_func(exp)
         ir.add_exported_function(self.module, f_id)
-        table.insert(cmds, ir.Cmd.NewClosure(exp.loc, dst, f_id))
+
+        local upvalues = {}
+        for _, upval in ipairs(func.captured_vars) do
```

Comment:
If we did that, I'm curious as to how we'd get the type information when emitting C.

Reply:
If the we have an `ir.Value.LocalVar` or an `ir.Value.Upvalue`we can look up the type in the `func.vars` and the `func.captured_vars`, respectively. Other kinds of  `ir.Value` would not give use the type information, but they should never happen so it's ok to `error("impossible")`.

Reply:
> If the we have an ir.Value.LocalVar or an ir.Value.Upvaluewe can look up the type in the func.vars and the func.captured_vars,

If it's a `LocalVar`, we can perform a lookup on `func.vars`, but if it's an upvalue, then how does a lookup in `captured_vars` help? Since `captured_vars` itself will `ir.Value`.  Do we perform some sort of recursive lookup until we get to the local var containing the `decl`?